NitroxModel/Serialization/PropertiesSerializer.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using NitroxModel.Logger;
+
+namespace NitroxModel.Serialization
+{
+    public static class PropertiesSerializer
+    {
+        private static readonly Dictionary<Type, Dictionary<string, MemberInfo>> typeCache = new Dictionary<Type, Dictionary<string, MemberInfo>>();
+
+        public static T Deserialize<T>() where T : IProperties, new()
+        {
+            T props = new T();
+            if (!File.Exists(props.FileName))
+            {
+                return props;
+            }
+
+            Dictionary<string, MemberInfo> typeCachedDict = GetTypeCacheDictionary<T>();
+            using StreamReader reader = new StreamReader(new FileStream(props.FileName, FileMode.Open), Encoding.UTF8);
+
+            char[] lineSeparator = { '=' };
+            int lineNum = 0;
+            string readLine;
+            while ((readLine = reader.ReadLine()) != null)
+            {
+                lineNum++;
+                if (readLine.Length < 1 || readLine[0] == '#')
```

Comment:
Would be nice but it significantly increases parsing complexity because string values need to be quoted and have support for escaping quotes too.

```
# Should the value be "hello " or "hello"?
myprop=helllo # some comment
# Hard to parse because need to allow "some\" value" too
other="world" # my comment
```